app/scripts/metamask-controller.js

```diff
@@ -4365,4 +4349,54 @@ export default class MetamaskController extends EventEmitter {
 
     return this.keyringController.setLocked();
   }
+
+  removePermissionsFor = (subjects) => {
+    try {
+      this.permissionController.revokePermissions(subjects);
+    } catch (exp) {
+      if (!(exp instanceof PermissionsRequestNotFoundError)) {
+        throw exp;
+      }
+    }
+  };
+
+  rejectPermissionsRequest = (requestId) => {
+    try {
+      this.permissionController.rejectPermissionsRequest(requestId);
+    } catch (exp) {
+      if (!(exp instanceof PermissionsRequestNotFoundError)) {
+        throw exp;
+      }
+    }
+  };
+
+  acceptPermissionsRequest = (request) => {
+    try {
+      this.permissionController.acceptPermissionsRequest(request);
+    } catch (exp) {
+      if (!(exp instanceof PermissionsRequestNotFoundError)) {
+        throw exp;
+      }
+    }
+  };
+
+  resolvePendingApproval = (id, value) => {
+    try {
+      this.approvalController.accept(id, value);
+    } catch (exp) {
+      if (!(exp instanceof ApprovalRequestNotFoundError)) {
+        throw exp;
+      }
+    }
+  };
+
+  rejectPendingApproval = (id, value) => {
+    try {
+      this.approvalController.reject(id, value);
```

Comment:
Based on the [ApprovalController](https://github.com/MetaMask/controllers/blob/5f9f5cee391f64e523ee4b1547ce29499b6f5385/src/approval/ApprovalController.ts), I think we'd want to change `value` here to `error`. 

Do we want to keep the same error format from before? 
``` 
new EthereumRpcError(error.code, error.message, error.data)
```
else, why was this removed?

Reply:
I renamed the variable.

Reply:
Fixed